shell_sort.c: Added Shell sort algorithm in C

bihanviranga opened the pull request:
This implementation uses half of the array's size as the gap.
Then it keeps dividing it by 2.

Closes [#135](https://github.com/NITSkmOS/Algorithms/issues/135)  

<!--
Thanks for your contribution!

Please take a quick look at those things down there. They're quite important.
Really! We wrote them for you. Yes you! With utmost care. Read them.

In order to help us evaluate PRs better, we ask you to have a look at the
following declaration and check the points you agree with. ( [x] )
PRs which don't agree to all the points mentioned below will be rejected.

Also make sure that this PR corresponds to one and only one issue. Write the
issue number in the space provided below.
-->

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #{ISSUE_NUMBER}


**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

sangamcse commented on shell_sort/C/shell_sort.c:
```diff
@@ -0,0 +1,30 @@
+
+void shellsort(int size, int *arr) {
+    // start with a big gap and reduce it
+    for(int gap = size/2; gap > 0; gap = gap/2) {
```
Missing space before ( in for(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse replied:
Missing space before ( in for(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse commented on shell_sort/C/shell_sort.c:
```diff
@@ -0,0 +1,30 @@
+void shellsort(int size, int *arr) {
+    // start with a big gap and reduce it
+    for(int gap = size/2; gap > 0; gap = gap/2) {
+        for(int i = gap; i < size; i = i+1) {
```
Missing space before ( in for(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse replied:
Missing space before ( in for(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse commented on shell_sort/C/shell_sort.c:
```diff
@@ -0,0 +1,30 @@
+
+            int j;
+            // shifting elements until the location for arr[i] is found
+            for(j = i; j >= gap && arr[j-gap] > temp; j = j-gap) {
```
Missing space before ( in for(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse replied:
Missing space before ( in for(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse commented on shell_sort/C/shell_sort.c:
```diff
@@ -0,0 +1,30 @@
+    int arr[6] = {10, 9, 8, 7, 6, 5 };
+    shellsort(arr_size, arr);
+
+    for(int i = 0; i < arr_size; i++) {
```
Missing space before ( in for(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse replied:
Missing space before ( in for(  [whitespace/parens] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

bihanviranga commented:
I added this to the README table.

sangamcse commented:
@sangamcse rebase

sangamcse commented:
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

sangamcse commented:
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

sangamcse commented:
Remind me once travis checks pass @bihanviranga

sangamcse commented:
Comment on a764675c63246f7a3522f8015760c27e422790d8, file queue/cpp/Queue.h, line 0.

No #ifndef header guard found, suggested CPP variable is: QUEUE_CPP_QUEUE_H_  [build/header_guard] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse commented:
Comment on a764675c63246f7a3522f8015760c27e422790d8, file queue/cpp/Queue.h, line 8.

private: should be indented +1 space inside class Queue  [whitespace/indent] [3]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse commented:
Comment on a764675c63246f7a3522f8015760c27e422790d8, file queue/cpp/Queue.h, line 10.

public: should be indented +1 space inside class Queue  [whitespace/indent] [3]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse commented:
Comment on cbc8dc110105ffdd11f045cfdb968a33b913c3b9, file queue/cpp/Queue.h, line 0.

No #ifndef header guard found, suggested CPP variable is: QUEUE_CPP_QUEUE_H_  [build/header_guard] [5]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse commented:
Comment on cbc8dc110105ffdd11f045cfdb968a33b913c3b9, file queue/cpp/Queue.h, line 8.

private: should be indented +1 space inside class Queue  [whitespace/indent] [3]

*Origin: CPPLintBear, Section: `all.cpp`.*

sangamcse commented:
Comment on cbc8dc110105ffdd11f045cfdb968a33b913c3b9, file queue/cpp/Queue.h, line 10.

public: should be indented +1 space inside class Queue  [whitespace/indent] [3]

*Origin: CPPLintBear, Section: `all.cpp`.*

bihanviranga commented:
@sangamcse Could you please tell me what to do? I replaced tabs with spaces as the message said. But the build isn't passing. Also the error messages mention a Queue.h which isn't a part of this PR. Also, I have updated the README file.

sangamcse commented:
@bihanviranga you need to rebase your repo as your repo is out of sync with original repo
https://github.com/NITSkmOS/Algorithms/blob/master/GIT_BASICS.md#rebasing

bihanviranga commented:
@sangamcse Thanks for the advice. Travis checks are passing now.
